tests/components/enphase_envoy/test_binary_sensor.py

```diff
@@ -0,0 +1,62 @@
+"""Test Enphase Envoy binary sensors."""
+
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.enphase_envoy import DOMAIN
+from homeassistant.components.enphase_envoy.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from tests.common import ConfigType, MockConfigEntry
+
+
+@pytest.fixture(name="setup_enphase_envoy_binary_sensor")
+async def setup_enphase_envoy_binary_sensor_fixture(
+    hass: HomeAssistant, config: ConfigType, mock_envoy: AsyncMock
+):
+    """Define a fixture to set up Enphase Envoy with binary sensor platform only."""
+    with (
+        patch(
+            "homeassistant.components.enphase_envoy.Envoy",
+            return_value=mock_envoy,
+        ),
+        patch(
+            "homeassistant.components.enphase_envoy.PLATFORMS",
+            [Platform.BINARY_SENSOR],
+        ),
+    ):
+        assert await async_setup_component(hass, DOMAIN, config)
+        await hass.async_block_till_done()
+        yield
+
+
+async def test_binary_sensor(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    setup_enphase_envoy_binary_sensor,
+) -> None:
+    """Test enphase_envoy binary_sensor entities."""
+
+    # number entities states should be created from test data
+    assert len(hass.states.async_all()) == 4
+
+    entity_registry = er.async_get(hass)
+    assert entity_registry
+
+    # compare registered entities against snapshot of prior run
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
+    )
+    assert entity_entries
+    assert entity_entries == snapshot
```

Comment:
I think it's good to have one snapshot entry per entity. With this you avoid a large diff when updating an entity, since this is an (unordered) list, adding an entity can make the list move all over the place, making the diff useless. The snapshot test in `flexit_bacnet` and `analytics_insight` is a good example.

Reply:
Like that concept, the large ones don't really help finding what changes. Implemented.